core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -209,7 +209,12 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                 builder.boost(nodeFloatValue(propNode));
                 iterator.remove();
             } else if (propName.equals("index_options")) {
-                builder.indexOptions(nodeIndexOptionValue(propNode));
+                if (builder.allowsIndexOptions()) {
+                    builder.indexOptions(nodeIndexOptionValue(propNode));
+                } else {
+                    throw new MapperParsingException(
```

Comment:
I will open a separate PR with deprecation logging instead of the error against the 6.x branches.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/26672 for the deprecation.